### PR TITLE
SOLR-16660 Deprecate analytics component

### PR DESCRIPTION
**Pull request by janhoy**
https://issues.apache.org/jira/browse/SOLR-16660

### CODE REVIEW

**Review by dsmiley — APPROVED** [review]

**Review by mkhludnev — APPROVED** [review]

File: solr/solr-ref-guide/modules/query-guide/pages/analytics-mapping-functions.adoc [hunk 1]
```diff
@@ -18,6 +18,8 @@
 
 Mapping functions map values for each Solr Document or Reduction.
 
+WARNING: The analytics component is deprecated. We recommend looking into similar functionality found in xref:json-facet-api.adoc[JSON Facet API].
```
Comment (HoustonPutman) [inline solr/solr-ref-guide/modules/query-guide/pages/analytics-mapping-functions.adoc:21]: Maybe we should link to JIRA and say create a ticket if there's functionality you need that isn't covered by JSON facets?

**Review by HoustonPutman — APPROVED** [review]
I think this is good. One possible addition.

File: solr/solr-ref-guide/modules/query-guide/pages/analytics-mapping-functions.adoc (hunk 1 above)
Comment (janhoy) [reply solr/solr-ref-guide/modules/query-guide/pages/analytics-mapping-functions.adoc:21]: Yea, or ask them to write to users list, where we can qualify that there is in fact a need? Jira is invite-only so not a good experience for users to end up there and be able to do nothing 😕. 

Feel free to suggest some text and I’ll incorporate it.

Comment (janhoy) [inline solr/solr-ref-guide/modules/query-guide/pages/analytics-mapping-functions.adoc:21]:
```suggestion
WARNING: The analytics component is deprecated. We recommend looking into similar functionality found in xref:json-facet-api.adoc[JSON Facet API]. Please notify the project if there's functionality you need that isn't currently covered by JSON facets.
```

Comment (HoustonPutman) [reply solr/solr-ref-guide/modules/query-guide/pages/analytics-mapping-functions.adoc:21]: Looks good to me, thanks for doing it!

**Pull request merged**